api/app/signals/apps/api/views/attachment.py

```diff
@@ -61,3 +59,27 @@ def get_signal(self):
         pk = self.kwargs.get('parent_lookup__signal__pk')
         signal = get_object_or_404(Signal.objects.filter_for_user(self.request.user), pk=pk)
         return signal
+
+    def destroy(self, *args, **kwargs):
+        user = self.request.user
+        signal = self.get_signal()
+        attachment = self.get_object()
+
+        if signal.is_parent and not user.has_perm('signals.sia_delete_attachment_of_parent_signal'):
+            msg = 'Cannot delete attachment need "sia_delete_attachment_of_parent_signal" permission.'
+            raise PermissionDenied(msg)
+        elif signal.is_child and not user.has_perm('signals.sia_delete_attachment_of_child_signal'):
+            msg = 'Cannot delete attachment need "sia_delete_attachment_of_child_signal" permission.'
+            raise PermissionDenied(msg)
+        elif (
+                not signal.is_parent and
+                not signal.is_child and
+                not user.has_perm('signals.sia_delete_attachment_of_normal_signal')):
+            msg = 'Cannot delete attachment need "sia_delete_attachment_of_normal_signal" permission.'
+            raise PermissionDenied(msg)
+
+        if attachment.created_by != user.email and not user.has_perm('signals.sia_delete_attachment_of_other_user'):
+            msg = 'Cannot delete attachment need "sia_delete_attachment_of_other_user" permission.'
+            raise PermissionDenied(msg)
+
+        return super().destroy(*args, **kwargs)
```

Comment:
A note should be made stating that an attachment has been deleted. This way it is visible in the history that there was an attachment and that it was deleted on a certain date by a logged-in user. Also add this to the tests so that we know a note has been made when a attachment has been deleted.

Reply:
Added